#! /usr/bin/env python

PACKAGE = "infobot_find_viewpoint"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# http://openkinect.org/wiki/Imaging_Information
gen.add("horizontal_angle_of_view",  double_t, 1, "The horizontal angle of view in radians.", 1.01229097, 0, 6.28318531)  # xtion camera HFOV is 58deg
gen.add("vertical_angle_of_view",    double_t, 1, "The vertical angle of view in radians.", 0.785398163, 0, 6.28318531)  # xtion camera HFOV is 45deg
gen.add("depth_max",                 double_t, 1, "The raytracing maximum distance.", 10.0, 0)
gen.add("fov_grid_res",              int_t,    1, "The FOV is discretized to a fov_grid_res x fov_grid_res grid.", 100, 0)
gen.add("dist_factor_a",             double_t, 1, "vis_prob = -1.0 * dist_fact_a * abs( dist - dist_fact_b ) - 1.0 * angle_factor * ang_dist + 1.0", 0.0, 0.0, 1.0)
gen.add("dist_factor_b",             double_t, 1, "vis_prob = -1.0 * dist_fact_a * abs( dist - dist_fact_b ) - 1.0 * angle_factor * ang_dist + 1.0", 0.0, 0.0, 1000.0)
gen.add("ang_factor",                double_t, 1, "vis_prob = -1.0 * dist_fact_a * abs( dist - dist_fact_b ) - 1.0 * angle_factor * ang_dist + 1.0", 0.0, 0.0, 1000.0)
gen.add("height_mean",               double_t, 1, "The pmap_to_octomap HEIGHT mode parameter.", 1.65, 0)  # average human height?
gen.add("height_var",                double_t, 1, "The pmap_to_octomap HEIGHT mode parameter.", 0.05, 0)

exit(gen.generate(PACKAGE, "infobot_find_viewpoint", "InfoBotFindViewpointVis"))
